receipt_dynamo/receipt_dynamo/entities/compaction_lock.py

```
@@ -107,21 +125,40 @@ def __repr__(self) -> str:
+    
+    collection_value = pk_parts[1]
+    lock_id = "#".join(pk_parts[2:])  # Rejoin in case lock_id contains #
+
+    # Validate collection value
+    try:
+        collection = ChromaDBCollection(collection_value)
+    except ValueError:
+        valid_values = [c.value for c in ChromaDBCollection]
+        raise ValueError(
+            f"Invalid collection in item: {collection_value}. Must be one of: {valid_values}"
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)